src/Integration.Vsix.UnitTests/CFamily/CFamilyEmbeddedSonarWayRulesTests.cs

```diff
@@ -31,29 +33,34 @@ namespace SonarLint.VisualStudio.Integration.Vsix.CFamily.UnitTests
     public class CFamilyEmbeddedSonarWayRulesTests
     {
         // Sanity checks that the rules metata for the CFamily plugin is present and can be loaded
+        private static int Active_C_Rules;
+        private static int Active_CPP_Rules;
 
-        // Note: how to find the expected number of active/inactive rules in SonarWay by language:
-        // 1. Start a local SQ instance with the correct plugin version installed
-        // 2. Browse to "Rules" e.g. http://localhost:9000/
-        // 3. Filter by Repository = SonarAnalyzer C
-        // 4. Filter by Quality Profile = Sonar way C
-        // The QP filter has "active/inactive" tabs. The number of rules is shown in the top-right of the screen.
-        // 5. Repeat for C++.
+        private readonly CFamilySonarWayRulesConfigProvider rulesMetadataCache = new CFamilySonarWayRulesConfigProvider(CFamilyShared.CFamilyFilesDirectory);
 
-        // Rule data for C-Family plugin v6.6 (build 13759)
-        private const int Active_C_Rules = 165;
-        private const int Inactive_C_Rules = 107;
+        [ClassInitialize]
+        public static void ClassInitialize(TestContext testContext)
+        {
+            Active_C_Rules = LoadRulesCount("c");
+            Active_CPP_Rules = LoadRulesCount("cpp");
+        }
 
-        private const int Active_CPP_Rules = 268;
-        private const int Inactive_CPP_Rules = 171;
+        private static int LoadRulesCount(string language)
+        {
+            using var client = new HttpClient();
+            var uri = $"https://sonarcloud.io/api/qualityprofiles/backup?qualityProfile=Sonar%20way&language={language}&organization=microsoft";
```

Comment:
Why `microsoft` rather than an org we own/control?

Reply:
Because then we'd have to be authenticated, as our org is Private. It doesn't matter much IMO as the profile is the same for everyone.

Reply:
Worth commenting, as it isn't obvious.